File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int start = 0;
        int end = height.size()-1;
        int maxArea = min(height[start],height[end])*abs(start-end); // calculating the area in first to last wall
        while(start < end){
            if(height[end] > height[start]){
                start++;
                maxArea = max(maxArea,min(height[start],height[end])*abs(start-end));
            } else {
                end--;
                maxArea = max(maxArea,min(height[start],height[end])*abs(start-end));
            }
        }

        return maxArea;
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l,r = 0,len(height)-1
        max_h = min(height[l],height[r]) * (r-l)
        print(max_h)
        while(l < r):
            if(height[l] < height[r]):
                l+=1
            else:
                r-=1
            max_h = max(max_h, min(height[l],height[r]) * (r-l))
        
        return max_h
            